com.microsoft.mrtk.input/Visualizers/ControllerVisualizer/ControllerModelLoader.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using UnityEngine;
+using System.Threading.Tasks;
+
+#if MROPENXR_PRESENT && (UNITY_STANDALONE_WIN || UNITY_WSA || UNITY_ANDROID) && GLTFAST_PRESENT && KTX_PRESENT
+using Microsoft.MixedReality.OpenXR;
+using GLTFast;
+#endif // MROPENXR_PRESENT && (UNITY_STANDALONE_WIN || UNITY_WSA || UNITY_ANDROID) && GLTFAST_PRESENT && KTX_PRESENT
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// A helper class which loads and caches controller models fetched from the controller's platform sdk
+    /// </summary>
+    public class ControllerModelLoader
+    {
+        /// <summary>
+        /// A dictionary which caches the controller model gameobject associated with a specified model key
+        /// </summary>
+        public static Dictionary<ulong, GameObject> ControllerModelDictionary = new Dictionary<ulong, GameObject>();
+
+        /// <summary>
+        /// Tries to load the controller model game object from the provided OpenXR ControllerModel.
+        /// Requires the MR OpenXR plugin to work.
+        /// </summary>
+        /// <param name="controllerModelProvider">The OpenXR ControllerModel to loade from</param>
+        /// <returns>A gameobject representing the generated controller model in the scene</returns>
+        public async static Task<GameObject> TryGenerateControllerModelFromPlatformSDK(Handedness handedness)
```

Comment:
I think this will produce a build warning if `#if MROPENXR_PRESENT && (UNITY_STANDALONE_WIN || UNITY_WSA || UNITY_ANDROID) && GLTFAST_PRESENT && KTX_PRESENT` resolves to `false`, since there will be no `await` in the `async` method body.

Reply:
Added `await Task.CompletedTask` in the alternative case to remove the warnings.